auth_fastapi/email_password.py

```diff
@@ -0,0 +1,159 @@
+            await _get_request_body(request)
+        )
+        sign_in_response = await email_password_client.sign_in(
+            sign_in_body.email, sign_in_body.password
+        )
+
+        match sign_in_response:
+            case email_password.SignInCompleteResponse():
+                _set_auth_cookie(sign_in_response.token_data.auth_token, response)
+            case _:
+                _set_verifier_cookie(sign_in_response.verifier, response)
+
+        return sign_in_response
+
+    async def handle_verify_email(
+        self,
+        request: Request,
+        response: Response,
+        verification_token: Annotated[str, Query()],
+        verifier: Annotated[Optional[str], Cookie(alias="edgedb_verifier")] = None,
+    ) -> email_password.EmailVerificationResponse:
+        email_password_client = await self.make_core()
+        return await email_password_client.verify_email(verification_token, verifier)
+
+    async def handle_send_password_reset(
+        self,
+        request: Request,
+        response: Response,
+    ) -> email_password.SendPasswordResetEmailResponse:
+        email_password_client = await self.make_core()
+        send_password_reset_body = email_password.SendPasswordResetBody.model_validate(
+            await _get_request_body(request)
+        )
+        send_password_reset_response = (
+            await email_password_client.send_password_reset_email(
+                send_password_reset_body.email
+            )
+        )
+
+        _set_verifier_cookie(send_password_reset_response.verifier, response)
+        return send_password_reset_response
+
+    async def handle_reset_password(
+        self,
+        request: Request,
+        response: Response,
+        verifier: Annotated[Optional[str], Cookie(alias="edgedb_verifier")] = None,
+    ) -> email_password.PasswordResetResponse:
+        email_password_client = await self.make_core()
+        password_reset_body = email_password.PasswordResetBody.model_validate(
+            await _get_request_body(request)
+        )
+        return await email_password_client.reset_password(
+            reset_token=password_reset_body.reset_token,
+            verifier=verifier,
+            password=password_reset_body.password,
+        )
+
+
+def make_email_password(
+    client: edgedb.AsyncIOClient, *, verify_url: str, reset_url: str
+) -> EmailPassword:
+    return EmailPassword(client=client, verify_url=verify_url, reset_url=reset_url)
+
+
+def _get_unchecked_exp(token: str) -> Optional[datetime.datetime]:
+    jwt_payload = jwt.decode(token, options={"verify_signature": False})
+    if "exp" not in jwt_payload:
+        return None
+    return datetime.datetime.fromtimestamp(jwt_payload["exp"], tz=datetime.timezone.utc)
+
+
+def _set_auth_cookie(token: str, response: Response) -> None:
+    exp = _get_unchecked_exp(token)
+    response.set_cookie(
+        key="edgedb_auth_token",
```

Comment:
We probably want this customizable? Or the whole `_set_auth_cookie()` function as a replaceable hook?